packages/js-client/test/integration/decoding.test.ts

```diff
@@ -0,0 +1,21 @@
+import { SimpleStorageClient, SimpleStorageContext } from '../../src';
+import * as ganacheSetup from '../helpers/ganache-setup';
+import { Server } from 'ganache';
+
+describe('Decoding', () => {
+  let server: Server;
+  beforeAll(async () => {
+    server = await ganacheSetup.start();
+    // deploy contracts and do other setup if necessary
+  });
+
+  afterAll(async () => {
+    server.close();
+  });
+
+  it('should decode an action', async () => {
+    const ctx = new SimpleStorageContext();
+    const client = new SimpleStorageClient(ctx);
+    client.decoding.myAction();
```

Comment:
Does this function still exist? Tests should be failing here

Reply:
nope, tests fixes are on the way

Reply:
It's a good practice to prevent PR's from getting too big, bit in this case we are shipping the initial commit. Personally I would prefer having a good initial version instead creating many more PR's to amend easy wins that are in our mind now